integration-tests/utils/integration-utils.js

```diff
@@ -211,9 +212,45 @@ const dismissNPSSurvey = async page => {
   }
 }
 
+// Test workaround: Retry loading of Terra UI if fails first time. This issue often happens after new deploy to Staging/Alpha.
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, navOptionNetworkIdle())
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
+  console.log('signIntoTerra ...')
+
+  const retryOptions = {
+    factor: 1,
+    minTimeout: 1000, // This is min wait time between retries
+    onFailedAttempt: error => {
+      console.error(
+        `Sign in to Terra attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`
+      )
+    },
+    retries: 2
+  }
+
+  const timeout = 60 * 1000
+  const run = async url => {
+    try {
+      const httpResponse = await page.goto(url, { waitUntil: ['networkidle0', 'domcontentloaded'], timeout })
+      if (httpResponse.status() >= 400) {
+        throw new Error(`Error loading URL: ${url}. Http response status: ${httpResponse.statusText()}`)
+      }
+      await page.waitForFunction(url => {
+        return window.location.href.includes(url)
+      }, {}, url)
+      await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout })
+    } catch (e) {
+      console.error(e)
+      // Stop page loading, as if you hit "X" in the browser. ignore exception.
+      await page._client.send('Page.stopLoading').catch(err => void err)
+      throw new Error(e)
+    }
+  }
+
+  if (!!testUrl) {
+    console.log(`Loading URL: ${testUrl}`)
+    await pRetry(() => run(testUrl), retryOptions)
+  }
+  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout })
```

Comment:
I don't think you want this since it is also at line 240 (unless you expect that `testUrl` might not be defined, in which case put this call in an else).

Reply:
There are few tests that do not pass in `testUrl`. It's in `else` now.